src/swupd/hash_test.go

```diff
@@ -49,3 +50,34 @@ func TestHashPrinting2(t *testing.T) {
 		t.Errorf("in and out of hashtable do not match\n\t%v\n\t%v", sout, s)
 	}
 }
+
+func TestHashEqual(t *testing.T) {
+	someHashes := []struct {
+		hash string
+		val  hashval
+	}{
+		{"3a60eb03c76ce17f1d08e0b5844c0455f6136c9b4bd4dd54c98cad2783354635", 0},
+		{"b4b9333757d79e1e766dbb5db3160108e907e110bd19cba4d1d4230b299d0eb", 0},
+		{"99aff80fc35d08b36c69ed0340ea80805f0c1b81ba7c734db6434b29a24c8391", 0},
+	}
+	for i, tc := range someHashes {
+		// subtle point here, need to use the array index, rather than
+		// setting tc.val as tc is a copy of the entry, not a pointer to it
+		// See https://golang.org/ref/spec#RangeClause
+		someHashes[i].val = internHash(tc.hash)
```

Comment:
Do Hashes and invHash not need to be reset between each of these Test* functions that use internHash?

Reply:
They should reset.

Reply:
thanks for the tip on why a test I just wrote was panicking. I had reset Hashes but not invHash

Reply:
The only test which requires them to be reset is the first one, and that is only because it is expecting absolute indices. The other tests make no assumptions about what the values should be, and so don't care if there are existing values in the intern table or not. 

We happen to know that the values are indices into an array, but as far as users of this library are concerned they are just obscure values, and the tests reflect that.

Reply:
right, important to note that when they do need to be reset, they need to be reset *at the same time*